index.rst

```diff
@@ -1,51 +1,1166 @@
+
+.. code-block:: python
+
+    from lsst.sconsUtils import scripts
+    scripts.BasicSConscript.pybind11(['minimize'])
+
+.. note::
+
+    It is important to change ``scripts.BasicSConscript.python`` (which uses swig) to ``scripts.BasicSConscript.pybind11`` (which uses pybind11).
+
+We also need to tell Python to import the new modules in ``python/lsst/afw/math/mathLib.py``, where we add the line
+
+.. code-block:: python
+
+    from __future__ import absolute_import
+    from ._minimize import *
+
+Since we are wrapping the header file ``minimize.h`` we must make sure to include it in ``minimize.cc`` (which is the previously created pybind11 template):
+
+.. code-block:: c++
+
+    #include "lsst/afw/math/minimize.h"
+
+.. _wrap_struct:
+
+Wrapping a struct
+-----------------
+
+The header file ``minimize.h`` contains the following code:
+
+.. code-block:: c++
+
+    #include <memory>
+    #include "Minuit2/FCNBase.h"
+
+    #include "lsst/daf/base/Citizen.h"
+    #include "lsst/afw/math/Function.h"
+
+    namespace lsst {
+    namespace afw {
+    namespace math {
+
+        struct FitResults {
+        public:
+            bool isValid;   ///< true if the fit converged; false otherwise
+            double chiSq;   ///< chi squared; may be nan or infinite, but only if isValid false
+            std::vector<double> parameterList; ///< fit parameters
+            std::vector<std::pair<double,double> > parameterErrorList; ///< negative,positive (1 sigma?) error for each parameter
+        };
+
+        template<typename ReturnT>
+        FitResults minimize(
+            lsst::afw::math::Function1<ReturnT> const &function,
+            std::vector<double> const &initialParameterList,
+            std::vector<double> const &stepSizeList,
+            std::vector<double> const &measurementList,
+            std::vector<double> const &varianceList,
+            std::vector<double> const &xPositionList,
+            double errorDef
+        );
+
+        template<typename ReturnT>
+        FitResults minimize(
+            lsst::afw::math::Function2<ReturnT> const &function,
+            std::vector<double> const &initialParameterList,
+            std::vector<double> const &stepSizeList,
+            std::vector<double> const &measurementList,
+            std::vector<double> const &varianceList,
+            std::vector<double> const &xPositionList,
+            std::vector<double> const &yPositionList,
+            double errorDef
+        );
+
+    }}}   // lsst::afw::math
+
+    #endif // !defined(LSST_AFW_MATH_MINIMIZE_H)
+
+
+We notice that ``minimize`` is a function that returns type ``FitResults``, and since ``FitResults`` is an ordinary structure we will wrap it first.
+In ``minimize.cc``, ``PYBIND11_PLUGIN`` contains the code to initialize the Python module ``minimize``, and all of the methods will be placed in this structure.
```

Comment:
Yes, I tend to be a bit too loose with my language sometimes. Thanks for the correction.